lambdas/db-provision-user-database/src/index.ts

```diff
@@ -54,26 +56,35 @@ export const handler = async (event: HandlerEvent): Promise<void> => {
       }
     });
 
-    const userExistsResults = await userExists(dbUser, knex);
-    const dbExistsResults = await dbExists(`${dbUser}_db`, knex);
+    const userExistsResults = await userExists(dbUser, knexClient);
+    const dbExistsResults = await dbExists(`${dbUser}_db`, knexClient);
 
     if (userExistsResults.length === 0) {
-      await knex.raw(`create user "${dbUser}" with encrypted password '${event.dbPassword}'`);
+      await knexClient.raw(`create user "${dbUser}" with encrypted password '${event.dbPassword}'`);
     } else {
-      await knex.raw(`alter user "${dbUser}" with encrypted password '${event.dbPassword}'`);
+      await knexClient.raw(`alter user "${dbUser}" with encrypted password '${event.dbPassword}'`);
     }
 
     if (event.dbRecreation) {
       if (dbExistsResults.length !== 0) {
-        await knex.raw(`alter database "${dbUser}_db" connection limit 0;`);
-        await knex.raw(`select pg_terminate_backend(pg_stat_activity.pid) from pg_stat_activity where pg_stat_activity.datname = '${dbUser}_db'`);
-        await knex.raw(`drop database "${dbUser}_db";`);
+        await knexClient.raw(`alter database "${dbUser}_db" connection limit 0;`);
+        await knexClient.raw(`select pg_terminate_backend(pg_stat_activity.pid) from pg_stat_activity where pg_stat_activity.datname = '${dbUser}_db'`);
+        await knexClient.raw(`drop database "${dbUser}_db";`);
       }
-      await knex.raw(`create database "${dbUser}_db";`);
+      await knexClient.raw(`create database "${dbUser}_db";`);
     } else if (dbExistsResults.length === 0) {
-      await knex.raw(`create database "${dbUser}_db";`);
+      await knexClient.raw(`create database "${dbUser}_db";`);
     }
-    await knex.raw(`grant all privileges on database "${dbUser}_db" to "${dbUser}"`);
+
+    await knexClient.raw(`grant all privileges on database "${dbUser}_db" to "${dbUser}"`);
+
+    await knexClient.destroy();
+
+    // connect to user database
+    const userDbKnexConfig = cloneDeep(rootKnexConfig);
+    set(userDbKnexConfig, 'connection.database', `${dbUser}_db`);
+    knexClient = knex(userDbKnexConfig);
```

Comment:
Why are we directly using knex versus using getKnexClient?     

Also we should avoid mutating the knexClient and just set a different client name.     Possibly consider adminKnexClient/userKnexClient?